modules/programs/tmux.nix

```diff
@@ -277,6 +473,79 @@ in
         '';
       })
 
+      (mkIf (cfg.status != null) {
+        home.file.".tmux.conf".text = mkOrder 1000 ''
+          set -g status on
+
+          set -g status-interval ${builtins.toString cfg.status.updateInterval}
+          set -g status-justify ${cfg.status.justification}
+          set -g status-left-length ${builtins.toString cfg.status.leftLength}
+          set -g status-right-length ${builtins.toString cfg.status.rightLength}
+
+          ${optionalString (cfg.status.currentWindowFormat != "")
+                           "setw -g window-status-current-format '${cfg.status.currentWindowFormat}'"}
+          ${optionalString (cfg.status.windowFormat != "")
+                           "setw -g window-status-format '${cfg.status.windowFormat}'"}
+          ${optionalString (cfg.status.leftFormat != "")
+                           "set -g status-left '${cfg.status.leftFormat}'"}
+          ${optionalString (cfg.status.rightFormat != "")
+                           "set -g status-right '${cfg.status.rightFormat}'"}
+          ${optionalString (cfg.status.style != "")
+                           "set -g status-style '${cfg.status.style}'"}
+          ${optionalString (cfg.status.windowStyle != "")
+                           "setw -g window-status-style '${cfg.status.windowStyle}'"}
+          ${optionalString (cfg.status.currentWindowStyle != "")
+                           "setw -g window-status-current-style '${cfg.status.currentWindowStyle}'"}
+          ${optionalString (cfg.status.messageStyle != "")
+                           "setw -g message-style '${cfg.status.messageStyle}'"}
+        '';
+      })
+
+      (mkIf (cfg.status == null) {
+        home.file.".tmux.conf".text = mkOrder 1000 ''
+          set -g status off
+        '';
+      })
+
+      (mkIf (cfg.hooks != {}) {
+        home.file.".tmux.conf".text = mkOrder 1100 ''
+          ${builtins.concatStringsSep "\n\n"
+            (lib.mapAttrsToList (key: value: ''
+              set-hook -g ${key} "${value}"
+            '' cfg.hooks))}
+        '';
+      })
+
+      (mkIf (cfg.bindings != null) {
+        home.file.".tmux.conf".text = mkOrder 1200 ''
+          ${optionalString (cfg.bindings.copyMode != {})
+                           (builtins.concatStringsSep "\n"
+                             (lib.mapAttrsToList (key: command: ''
+                               bind -T copy-mode ${key} ${command}
+                             '') cfg.bindings.copyMode))}
+          ${optionalString (cfg.bindings.viCopyMode != {})
+                           (builtins.concatStringsSep "\n"
+                             (lib.mapAttrsToList (key: command: ''
+                               bind -T copy-mode-vi ${key} ${command}
```

Comment:
What if we only have `cfg.bindings.copyMode`, and use `cfg.keyMode` to decide between `copy-mode` and `copy-mode-vi`?

Reply:
Good idea, I think, it will reduce duplication.